sunpy/timeseries/sources/goes.py

```diff
@@ -89,7 +89,7 @@ def plot(self, axes=None, columns=None, **kwargs):
         axes, columns = self._setup_axes_columns(axes, columns)
         plot_settings = {"xrsa": ["blue", r"0.5--4.0 $\AA$"], "xrsb": ["red", r"1.0--8.0 $\AA$"]}
         data = self.to_dataframe()
-        for channel in columns:
+        for channel in ['xrsa', 'xrsb']:
```

Comment:
This is not the right fix. It ignores any columns specifed in the `columns` kwarg (as evidenced by the failing test).